account_invoice_inter_company/models/res_company.py

```diff
@@ -7,14 +7,17 @@ class ResCompany(models.Model):
 
     _inherit = 'res.company'
 
+    company_share_product = fields.Boolean(
+        'Share product to all companies', compute='_compute_share_product')
     invoice_auto_validation = fields.Boolean(
         help="When an invoice is created by a multi company rule "
              "for this company, it will automatically validate it",
         default=True)
-    use_inter_company_products = fields.Boolean(
-        help="Use the same products when an invoice is created by "
-             "a multi company rule.",
-        default=True)
+
+    def _compute_share_product(self):
+        self.ensure_one()
+        product_rule = self.env.ref('product.product_comp_rule')
```

Comment:
In case the multicompany product rule is another one, different than `product.product_comp_rule`, the feature would not be working. Can't the old mechanism based on field `use_inter_company_products` be restored, instead of using this computed boolean `company_share_product`? Or perhaps another solution would be making the multicompany product rule selectable in the configuration, in case is other than `product.product_comp_rule`.

I have some doubts about this point :thinking:

Reply:
@astirpe thanks for your remark

If you have defined another rule to deal with product rule, I think the better for your project is to put in your custom code this field as readonly to avoid to break the erp behavior.
In this case no side effect.
You also may override the compute method to be compatible with the right rule : the purpose is just to active/unactive rules, then you can just do it without method or settings

I think it's better to plug into Odoo new settings than continue with an extra field for the same feature.

Reply:
Of course I can override the compute methods and do whatever I want for my projects, but that's not the point. :wink: 

I was not thinking about adding an extra field, I was thinking about restoring the old mechanism and not using the `not bool(product_rule.active)` black magic at all. Could be confusing and is not even mentioned in the readme at the moment.

Reply:
Ok, I have no strong opinion here. 
I can mention in the readme, no problem
Let's see what others can think about that.

Please @pedrobaeza

Reply:
Well, I see the intention of @bealdav removing that extra field for avoiding incoherences between one company and the other, and also reducing the number of configuration options. There's no special problem on the "hardcoded" rule, as you shouldn't have another multi-company rule, having this one marked as `noupdate` and being able to change it.

@astirpe if I don't remember bad, you were the one that added the inter company products feature. Is the granularity by company really needed?

Reply:
> Is the granularity by company really needed?

Could be possible in theory, it depends on the relations and agreements between companies I guess.

But I would prefer some functional expert to say an opinion about this (as I said, I have doubts about this part). 
I'm not blocking the PR because of this point. But at least I would like to see the new proposed behavior of this module explained in the readme.